### ISSUE TITLE
Update AngleSharp to 0.10.0

### ISSUE THREAD

**Issue by NikolayIT** [mganss/HtmlSanitizer#154]
https://github.com/AngleSharp/AngleSharp/releases/tag/v0.10.0

Please keep in mind that the old .NET 4 and Silverlight targets are dropped. AngleSharp now targets .NET Standard 2.0.

Also you might need to install AngleSharp.Css package.

Migration guide: https://github.com/AngleSharp/AngleSharp/blob/master/doc/Migration.md

**Comment by mganss** [comment]
I've merged master into the AngleSharp_0_10 branch. It compiles but a large number of unit tests fail. AngleSharp.Css does some surprising "normalization" of CSS values, e.g.
```html
<div style="color: black; background-color: white">Test</div>
```
becomes
```html
<div style="color: rgba(0, 0, 0, 1); background: rgba(255, 255, 255, 1)">Test</div>
```
At first glance this behavior seems to contradict what browsers are doing. I'll try to dig deeper and perhaps open issues at AngleSharp.Css.

See also #117. I'll close that issue so discussion can continue here.

**Comment by stweb** [comment]
Hmm, I need .Net Standard compatibility which comes with AngleSharp 0.10.0 - could you provide a beta version which targets it? The we can see if the CSS changes are causing issues.

**Comment by mganss** [comment]
@stweb HtmlSanitizer today targets netstandard1.3 and netstandard2.0 (in addition to net40 and net45). After the update to AngleSharp it will only target netstandard2.0 and net46.

**Comment by NikolayIT** [comment]
Can you please publish 4.0.200 (as preview) in NuGet? :)

**Comment by mganss** [comment]
@NikolayIT AngleSharp.Css does not (yet?) support netstandard1.3. I will have to wait for a few issues in AngleSharp.Css to be resolved, in particular AngleSharp/AngleSharp.Css#14. What prevents you from using the current stable HtmlSanitizer?

**Comment by NikolayIT** [comment]
I don't have problems using the latest stable HtmlSanitizer. I just want to use the new version of AngleSharp and I need both HtmlSanitizer and AngleSharp in my solution :)

**Comment by 304NotModified** [comment]
> AngleSharp.Css does some surprising "normalization" of CSS values

That's bad :(

**Comment by tiesont** [comment]
> > AngleSharp.Css does some surprising "normalization" of CSS values
> 
> That's bad :(

@304NotModified  Agreed. If I have `background: #fff` as a style on some element, I expect that to be unchanged.

**Comment by mesutcakir** [comment]
![image](https://user-images.githubusercontent.com/4125460/53013321-8f72ad80-3456-11e9-8ae3-420367a87a13.png)

System.TypeLoadException: 'AngleSharp.Parser.Html.HtmlParser' türü 'AngleSharp, Version=0.11.0.0, Culture=neutral, PublicKeyToken=e83494dcdc6d31ea'

**Comment by farlock85** [comment]
@mesutcakir unfortunately you can't run HtmlSanitizer and have AngleSharp installed. I ran in the same issue today.

**Comment by 304NotModified** [comment]
> unfortunately you can't run HtmlSanitizer and have AngleSharp installed

Yes you can, but you can't mix versions. That's because anglesharp makes breaking changes. (Unfortunately allowed with Semver because it's not 1.0 yet)

**Comment by NikolayIT** [comment]
@mganss, Any updates on this? :-)

**Comment by mganss** [comment]
There has been no progress at AngleSharp.Css (still at 0.10.0). I will look into this again once a new version of AngleSharp.Css has been released that fixes https://github.com/AngleSharp/AngleSharp.Css/issues/14.

**Comment by mganss** [comment]
AngleSharp.Css 0.12.1 has been released which fixes AngleSharp/AngleSharp.Css#14. It doesn't fix https://github.com/AngleSharp/AngleSharp.Css/issues/9, however, and it seems this won't been fixed in future releases, either. The latter issue is the one where `color: black` (and all other color representations such as `#000`) get normalized to `rgba(0, 0, 0, 1)`. This also has an upside, though: You can always expect the same representation when matching CSS values in `DisallowCssPropertyValue`.

I've released 5.0.125-beta with dependencies on AngleSharp, AngleSharp.Css 0.12.1.

**Comment by 304NotModified** [comment]
Any Idea if this is possible ? From 
https://github.com/AngleSharp/AngleSharp.Css/issues/9

> Remark: Actually, also AngleSharp today is performing this kind of normalization, but it can be turned off via a simple flag. In AngleSharp.Css this is much more flexible and you just insert a different interface to handle it.

**Comment by mganss** [comment]
I think it's the same as "Write your own value creator" in that same comment (https://github.com/AngleSharp/AngleSharp.Css/issues/9#issuecomment-289161118). I asked specifically about this in https://github.com/AngleSharp/AngleSharp.Css/issues/9#issuecomment-452356939 but didn't get a response.

**Comment by NikolayIT** [comment]
Updating both AngleSharp to latest stable and HtmlSanitizer to latest preview is working as expected.

Thanks!

**Comment by jerone** [comment]
Please update AngleSharp to the latest.

**Comment by mganss** [comment]
@jerone The latest beta [5.0.218-beta](https://www.nuget.org/packages/HtmlSanitizer/5.0.218-beta) uses the latest AngleSharp version (0.12.1). I've been wanting to try dependabot, it's only been free for a couple of weeks now that they've been acquired by GitHub.

**Comment by 304NotModified** [comment]
Im using dependabot for weeks, works great! (But only with sdk style csprojs)

**Comment by mganss** [comment]
@304NotModified Yeah, very nice. It also automatically rebases PRs and automatically closes them if they're no longer needed. I enabled it for a good handful of projects at once and immediately got a ton of e-mails 🙄

**Comment by Doidel** [comment]
I'm also using AngleSharp alongside HtmlSanitizer. Thanks for those beta versions, saved my life! *phew*

**Comment by mganss** [comment]
Has been released in 5.0.298
